isisd/isis_pdu.c

```diff
@@ -187,8 +187,7 @@ static int process_p2p_hello(struct iih_info *iih)
 		adj->sys_type = ISIS_SYSTYPE_UNKNOWN;
 	}
 
-	if (tw_adj && adj->threeway_state == ISIS_THREEWAY_DOWN)
-		adj->ext_circuit_id = tw_adj->local_circuit_id;
+	adj->ext_circuit_id = tw_adj->local_circuit_id;
```

Comment:
Makes sense to me, but shouldn't we keep the `tw_adj` null check?

Reply:
Yes, you are indeed correct there. :+1: